### PR TITLE
docs: add badges to readme

### PR DESCRIPTION
**Pull request by austinvazquez**
Issue #, if available:
N/A

*Description of changes:*
Adds a set of badges to readme for quick view of Go packaging, CI, Benchmarks, and Go Report Card.

*Testing done:*
Manual

- [x] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ginglis13 — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -8,6 +8,13 @@
 ## Hello, Finch
 <!-- markdownlint-restore -->
 
+[![PkgGoDev](https://pkg.go.dev/badge/github.com/runfinch/finch)](https://pkg.go.dev/github.com/runfinch/finch)
+[![Go Report Card](https://goreportcard.com/badge/github.com/runfinch/finch)](https://goreportcard.com/report/github.com/runfinch/finch)
+[![CI](https://github.com/runfinch/finch/actions/workflows/ci.yaml/badge.svg?branch=main)](https://github.com/runfinch/finch/actions/workflows/ci.yaml)
+[![Static Badge](https://img.shields.io/badge/Website-Benchmarks-blue)](https://runfinch.github.io/finch/dev/bench/)
+[![GitHub release (latest by date)](https://img.shields.io/github/v/release/runfinch/finch?logo=GitHub)](https://github.com/runfinch/finch/releases)
+![GitHub all releases](https://img.shields.io/github/downloads/runfinch/finch/total?label=all%20time%20downloads)
```
Comment (ginglis13) [inline README.md:16]: nit if you want the "all time downloads" badge to be a link as well
```suggestion
[![GitHub all releases](https://img.shields.io/github/downloads/runfinch/finch/total?label=all%20time%20downloads)](https://img.shields.io/github/downloads/runfinch/finch/total?label=all%20time%20downloads)
```

Comment (austinvazquez) [reply README.md:16]: That would just link to the badge. Perhaps I can have the link go to https://github.com/runfinch/finch/releases?

Comment (sam-berning) [reply README.md:16]: I don't think it's necessarily an issue to have two links going to the same place, and I kinda like the flow of "wow, it's been installed a lot, how can I install it?" -> the release page. +1 in favor of linking to https://github.com/runfinch/finch/releases

Comment (austinvazquez) [reply README.md:16]: Makes sense. Updated to link to releases. PTAL.

**Review by ginglis13 — APPROVED** [review]

**Review by sam-berning — APPROVED** [review]

**Pull request merged**